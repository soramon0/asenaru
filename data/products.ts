import { Product } from '@/types/models'

export const products: Product[] = [
  { id: 1, name: 'Black Hat', price: 80, image: '/hat.png' },
  { id: 2, name: 'Lightwieght Jacket', price: 249.99, image: '/jacket.png' },
  { id: 3, name: 'T-Shirt', price: 160.12, image: '/t-shirt.png' },
]
