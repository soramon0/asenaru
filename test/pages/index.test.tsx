import React from 'react'
import { axe } from 'jest-axe'
import { render } from '@testUtils'
import { ImageProps } from 'next/image'

import { products } from '@/data/products'
import Home from '@/pages/index'

describe('Home page', () => {
  it('is accessible', async () => {
    const { container } = render(<Home products={products} />)
    const results = await axe(container)
    expect(results).toHaveNoViolations()
  })

  it('matches snapshot', () => {
    const { asFragment } = render(<Home products={products} />)
    expect(asFragment()).toMatchSnapshot()
  })
})

beforeAll(() => {
  Object.defineProperty(require('next/image'), 'default', {
    writable: true,
    value: (props: ImageProps) => <img {...props} />,
  })

  Object.defineProperty(window, 'IntersectionObserver', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
      observe: () => null,
      disconnect: () => null,
    })),
  })
})

afterAll(() => {
  jest.resetAllMocks()
})
