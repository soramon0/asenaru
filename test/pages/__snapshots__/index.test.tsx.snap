// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="flex flex-wrap md:h-[calc(100vh-80px)]"
  >
    <div
      class="md:w-2/3 h-full"
    >
      <article
        class="w-full h-full relative group overflow-hidden bg-violet"
      >
        <div
          class="absolute z-10 top-0 left-0"
        >
          <h3
            class="px-4 py-3 text-3xl font-bold tracking-wide bg-primary transition duration-500 ease-in-out group-hover:bg-violet group-hover:text-primary-2 md:px-8 md:py-5 md:text-2xl xl:px-6 xl:text-3xl"
            id="black-hat"
          >
            Black Hat
          </h3>
          <span
            class="inline-block px-4 py-3 font-medium tracking-wide text-center bg-primary transition duration-500 ease-in-out group-hover:bg-violet group-hover:text-primary-2 md:px-6 md:py-4 md:text-md"
          >
            80 USD
          </span>
        </div>
        <button
          class="absolute z-10 top-0 right-0 p-2 bg-primary transition duration-500 ease-in-out group-hover:bg-violet"
        >
          <svg
            aria-hidden="true"
            class="h-7 w-7 group-hover:text-primary-2"
            fill="none"
            height="24"
            stroke="currentColor"
            viewBox="0 0 24 24"
            width="24"
          >
            <path
              d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <a
          aria-labelledby="black-hat"
          class="w-full h-full group inline-block bg-pattern-product transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-[98%]"
          href="/product/1"
        >
          <div
            class="h-full w-full transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-110 group-focus:scale-110"
          >
            <div
              style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
            >
              <div
                style="box-sizing: border-box; display: block; max-width: 100%;"
              >
                <img
                  alt=""
                  aria-hidden="true"
                  role="presentation"
                  src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSIxMjgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                  style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                />
              </div>
              <noscript />
              <img
                alt="black lightwieght jacket"
                decoding="async"
                src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
              />
            </div>
          </div>
        </a>
      </article>
    </div>
    <div
      class="md:w-1/3 md:h-1/2"
    >
      <div
        class="md:h-full"
      >
        <article
          class="w-full h-full relative group overflow-hidden bg-secondary-2"
        >
          <div
            class="absolute z-10 top-0 left-0"
          >
            <h3
              class="px-4 py-3 text-3xl font-bold tracking-wide bg-primary transition duration-500 ease-in-out group-hover:bg-secondary-2 group-hover:text-primary-2 md:px-8 md:py-5 md:text-2xl xl:px-6 xl:text-3xl"
              id="lightwieght-jacket"
            >
              Lightwieght Jacket
            </h3>
            <span
              class="inline-block px-4 py-3 font-medium tracking-wide text-center bg-primary transition duration-500 ease-in-out group-hover:bg-secondary-2 group-hover:text-primary-2 md:px-6 md:py-4 md:text-md"
            >
              249.99 USD
            </span>
          </div>
          <button
            class="absolute z-10 top-0 right-0 p-2 bg-primary transition duration-500 ease-in-out group-hover:bg-secondary-2"
          >
            <svg
              aria-hidden="true"
              class="h-7 w-7 group-hover:text-primary-2"
              fill="none"
              height="24"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <a
            aria-labelledby="lightwieght-jacket"
            class="w-full h-full group inline-block bg-pattern-product transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-[98%]"
            href="/product/2"
          >
            <div
              class="h-full w-full transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-110 group-focus:scale-110"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSIxMjgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="black lightwieght jacket"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </a>
        </article>
      </div>
      <div
        class="md:h-full"
      >
        <article
          class="w-full h-full relative group overflow-hidden bg-pink"
        >
          <div
            class="absolute z-10 top-0 left-0"
          >
            <h3
              class="px-4 py-3 text-3xl font-bold tracking-wide bg-primary transition duration-500 ease-in-out group-hover:bg-pink group-hover:text-primary-2 md:px-8 md:py-5 md:text-2xl xl:px-6 xl:text-3xl"
              id="t-shirt"
            >
              T-Shirt
            </h3>
            <span
              class="inline-block px-4 py-3 font-medium tracking-wide text-center bg-primary transition duration-500 ease-in-out group-hover:bg-pink group-hover:text-primary-2 md:px-6 md:py-4 md:text-md"
            >
              160.12 USD
            </span>
          </div>
          <button
            class="absolute z-10 top-0 right-0 p-2 bg-primary transition duration-500 ease-in-out group-hover:bg-pink"
          >
            <svg
              aria-hidden="true"
              class="h-7 w-7 group-hover:text-primary-2"
              fill="none"
              height="24"
              stroke="currentColor"
              viewBox="0 0 24 24"
              width="24"
            >
              <path
                d="M4.318 6.318a4.5 4.5 0 000 6.364L12 20.364l7.682-7.682a4.5 4.5 0 00-6.364-6.364L12 7.636l-1.318-1.318a4.5 4.5 0 00-6.364 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </button>
          <a
            aria-labelledby="t-shirt"
            class="w-full h-full group inline-block bg-pattern-product transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-[98%]"
            href="/product/3"
          >
            <div
              class="h-full w-full transition-transform delay-150 duration-700 ease-in-out transform group-hover:scale-110 group-focus:scale-110"
            >
              <div
                style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
              >
                <div
                  style="box-sizing: border-box; display: block; max-width: 100%;"
                >
                  <img
                    alt=""
                    aria-hidden="true"
                    role="presentation"
                    src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI4MCIgaGVpZ2h0PSIxMjgwIiB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIHZlcnNpb249IjEuMSIvPg=="
                    style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
                  />
                </div>
                <noscript />
                <img
                  alt="black lightwieght jacket"
                  decoding="async"
                  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                />
              </div>
            </div>
          </a>
        </article>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
